php-frontend/src/main/java/org/sonar/php/tree/symbols/DeclarationVisitor.java

```diff
@@ -85,17 +87,16 @@ public class DeclarationVisitor extends NamespaceNameResolvingVisitor {
   private final Map<FunctionDeclarationTree, FunctionSymbolData> functionSymbolDataByTree = new HashMap<>();
   private FunctionSymbolIndex functionSymbolIndex;
 
-  private Deque<ClassTree> classTreeStack = new ArrayDeque<>();
-  private Deque<FunctionSymbolProperties> functionPropertiesStack = new ArrayDeque<>();
+  private final Deque<ClassTree> classTreeStack = new ArrayDeque<>();
+  private final Deque<FunctionSymbolProperties> functionPropertiesStack = new ArrayDeque<>();
 
   private static final Set<String> VALID_VISIBILITIES = SetUtils.immutableSetOf("PUBLIC", "PRIVATE", "PROTECTED");
   private static final QualifiedName ANONYMOUS_CLASS_NAME = QualifiedName.qualifiedName("<anonymous_class>");
 
-  DeclarationVisitor(SymbolTableImpl symbolTable, ProjectSymbolData projectSymbolData, @Nullable PhpFile file) {
+  DeclarationVisitor(SymbolTableImpl symbolTable, ProjectSymbolData projectSymbolData, PhpFile file) {
```

Comment:
`file` cannot be null anymore? I see that `pathOf` is still checking it.
Or do we put `@Nullable` notation on parameters only when it matters directly in the function itself?
I would still tend to keep it to show that even at this stage we can have `file == null`, in case of future modification where `pathOf` would not be there anymore, we would lose this important information.

Reply:
Nice catch. I was playing around with NotNull. I will re-add the annotation.